### PR TITLE
chore: make reset_gitlab() better

### PR DESCRIPTION
**Pull request by JohnVillalovos**
Saw issues in the CI where reset_gitlab() would fail. It would fail to
delete the group that is created when GitLab starts up. Extending the
timeout didn't fix the issue.

Changed the code to use the new `helpers.safe_delete()` function.
Which will delete the resource and then make sure it is deleted before
returning.

Also added some logging functionality that can be seen if logging is
turned on in pytest.

### CODE REVIEW

**Comment by JohnVillalovos** [comment]
@nejch Please look at the function testing logs and let me know what you think.

**Comment by JohnVillalovos** [comment]
An example of what this is fixing from https://github.com/python-gitlab/python-gitlab/runs/4656774636?check_suite_focus=true#step:5:27

```
tests/functional/api/test_clusters.py::test_project_clusters 
-------------------------------- live log setup --------------------------------
2021-12-29 08:13:06.617 [    INFO] (conftest.py:gl:L258) Create python-gitlab gitlab.Gitlab object
2021-12-29 08:13:07.660 [    INFO] (conftest.py:_reset_gitlab_delete_resources:L35) Mark for deletion project: 'Monitoring'
2021-12-29 08:13:08.277 [    INFO] (conftest.py:_reset_gitlab_delete_resources:L44) Mark for deletion group: 'GitLab Instance'
2021-12-29 08:13:08.684 [    INFO] (conftest.py:wait_for_maximum_list_length:L83) Iteration: 0: items in projects: []
2021-12-29 08:13:08.954 [    INFO] (conftest.py:wait_for_maximum_list_length:L83) Iteration: 0: items in groups: ['GitLab Instance']
# NOTE(jlvillal): Deleting again and no message about already marked for deletion, so this is a fix.
2021-12-29 08:13:08.955 [    INFO] (conftest.py:wait_for_maximum_list_length:L88) Marking again for deletion groups: 'GitLab Instance'
2021-12-29 08:13:09.142 [    INFO] (conftest.py:wait_for_maximum_list_length:L83) Iteration: 1: items in groups: ['GitLab Instance']
# NOTE(jlvillal): 2nd time through, deleting again but this time indicates already marked for deletion.
2021-12-29 08:13:09.143 [    INFO] (conftest.py:wait_for_maximum_list_length:L88) Marking again for deletion groups: 'GitLab Instance'
2021-12-29 08:13:09.177 [    INFO] (conftest.py:wait_for_maximum_list_length:L94) Already marked for deletion: 'GitLab Instance' 404: 404 Group Not Found
# NOTE(jlvillal): 3rd time through and group is now gone.
2021-12-29 08:13:09.308 [    INFO] (conftest.py:wait_for_maximum_list_length:L83) Iteration: 2: items in groups: []
2021-12-29 08:13:09.359 [    INFO] (conftest.py:wait_for_maximum_list_length:L83) Iteration: 0: items in variables: []
2021-12-29 08:13:09.525 [    INFO] (conftest.py:wait_for_maximum_list_length:L83) Iteration: 0: items in users: ['Administrator']
```

**Comment by JohnVillalovos** [comment]
@nejch By using `safe_delete()` it simplified things 🙂

File: tests/functional/conftest.py [hunk 1]
```diff
@@ -17,68 +17,37 @@ def fixture_dir(test_dir):
...
+    """Delete resources (such as projects, groups, users) that shouldn't
+    exist."""
     for project in gl.projects.list():
-        logging.info(f"Marking for deletion project: {project.path_with_namespace!r}")
         for deploy_token in project.deploytokens.list():
             logging.info(
-                f"Marking for deletion token: {deploy_token.username!r} in "
+                f"Deleting deploy token: {deploy_token.username!r} in "
                 f"project: {project.path_with_namespace!r}"
             )
```
Comment (nejch) [inline tests/functional/conftest.py:28]: I think we're now logging with info both in and outside the function - could bring it together at some point.

**Comment by nejch** [comment]
> @nejch By using `safe_delete()` it simplified things slightly_smiling_face

Amazing, love to see code get deleted :sweat_smile:

**Pull request merged**